src/components/Registration/Registration.tsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import { Button } from 'src/common/Button/Button';
+import Input from 'src/common/Input/Input';
+import * as text from 'src/constants';
+import { Link, Navigate } from 'react-router-dom';
+
+interface RegistrationState {
+	name: string;
+	email: string;
+	password: string;
+}
+
+class Registration extends React.Component<object, RegistrationState> {
+	constructor(props: object) {
+		super(props);
+		this.state = {
+			name: '',
+			email: '',
+			password: '',
+		};
+	}
+
+	registrationDone = false;
+
+	forbiddenSymbols = /[@#$%^&]/;
+
+	url = 'http://localhost:4000/register';
+
+	submit = function () {
+		console.log(this.state);
+		alert('Do register stuff!!!...');
+	};
+
+	sendRequest = function () {
```

Comment:
You can also use `async / await` when you work with Promises.

```suggestion
	sendRequest = async () => {
	  try {
	    const res = await fetch(...);
	    const body = await res.json();
	    ...
	  }
	  catch (err) {
	    ...
	  }
	}
```